src/admin/components/ResourceTabs/PolygonReviewTab/components/PolygonDrawer/components/VersionHistory.tsx

```diff
@@ -46,11 +54,74 @@ const VersionHistory = ({
   const t = useT();
   const { displayNotification } = useAlertHook();
   const { openModal, closeModal } = useModalContext();
+  const ctx = useShowContext();
+  const [files, setFiles] = useState<UploadedFile[]>([]);
+  const [saveFlags, setSaveFlags] = useState<boolean>(false);
 
   useEffect(() => {
     refetch();
   }, [selectPolygonVersion]);
 
+  useEffect(() => {
+    if (files && files.length > 0 && saveFlags) {
+      uploadFiles();
+      setSaveFlags(false);
+    }
+  }, [files, saveFlags]);
+
+  const getFileType = (file: UploadedFile) => {
+    const fileType = file?.file_name.split(".").pop()?.toLowerCase();
+    if (fileType === "geojson") return "geojson";
+    if (fileType === "zip") return "shapefile";
+    if (fileType === "kml") return "kml";
```

Comment:
you can have something like `return [geojson, zip, kml].includes(fileType) ?  fileType : null`